runtime/compiler/z/codegen/J9S390PrivateLinkage.cpp

```diff
@@ -2823,6 +2823,73 @@ void TR::J9S390JNILinkage::checkException(TR::Node * callNode,
    generateS390LabelInstruction(self()->cg(), TR::InstOpCode::LABEL, callNode, exceptionRestartLabel);
    }
 
+/**
+ * \brief JNI return value processing:
+ * 1) Unwrap return value for objects or
+ * 2) Enforce a return value of 0 or 1 for boolean return type.
+*/
+void
+TR::J9S390JNILinkage::processJNIReturnValue(TR::Node * callNode,
+                                            TR::CodeGenerator* cg,
+                                            TR::Register* javaReturnRegister,
+                                            TR::RegisterDependencyConditions* deps)
+   {
+   auto resolvedMethod = callNode->getSymbol()->castToResolvedMethodSymbol()->getResolvedMethod();
+   auto returnType = resolvedMethod->returnType();
+   bool isUnwrapAddressReturnValue = !((TR_J9VMBase *)fe())->jniDoNotWrapObjects(resolvedMethod)
+                                     && (returnType == TR::Address)
+                                     && (javaReturnRegister!= NULL)
+                                     && (javaReturnRegister->getKind() == TR_GPR);
```

Comment:
Why do we need to check both the `returnType` and `javaReturnRegister->getKind()`? If the `returnType` is an address we should be guaranteed for it to be in a GPR.

Reply:
Right. This is moved from the old source and I didn't take a second look at this. I'll remove the GPR check.